`make docs` fails with docutils 0.18

`make docs` fails with docutils 0.18:

```
$ make docs
sphinx-build -W doc/source doc/build/sphinx
Running Sphinx v1.4.4
mocking modules [] and types []
loading pickled environment... not yet created
loading intersphinx inventory from https://docs.python.org/objects.inv...
intersphinx inventory has moved: https://docs.python.org/ -> https://docs.python.org/3
[autosummary] generating autosummary for: administration.rst, api/remoteappmanager.application.rst, api/remoteappmanager.cli.remoteappdb.__main__.rst, api/remoteappmanager.cli.remoteapprest.__main__.rst, api/remoteappmanager.command_line_config.rst, api/remoteappmanager.db.csv_db.rst, api/remoteappmanager.db.interfaces.rst, api/remoteappmanager.db.orm.rst, api/remoteappmanager.docker.async_docker_client.rst, api/remoteappmanager.docker.container.rst, ..., design.rst, developer.rst, developer/ci_instructions.rst, developer/docker.rst, index.rst, troubleshoot/database.rst, troubleshoot/docker.rst, troubleshoot/general.rst, troubleshooting.rst, utilities.rst
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 40 source files that are out of date
updating environment: 40 added, 0 changed, 0 removed
reading sources... [  5%] api/remoteappmanager.application                                                                             
Exception occurred:
  File "/home/pzahemszky/Repositories/simphony-remote/venv/lib/python3.6/site-packages/sphinx/util/nodes.py", line 51, in apply_source_workaround
    for classifier in reversed(node.parent.traverse(nodes.classifier)):
TypeError: 'generator' object is not reversible
The full traceback has been saved in /tmp/sphinx-err-pzprrroh.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
Makefile:218: recipe for target 'docs' failed
make: *** [docs] Error 1
```

<details>

<summary>Contents of /tmp/sphinx-err-pzprrroh.log</summary>

```
# Sphinx version: 1.4.4
# Python version: 3.6.9 (CPython)
# Docutils version: 0.18 release
# Jinja2 version: 3.0.3
# Last messages:
#   not yet created
#   loading intersphinx inventory from https://docs.python.org/objects.inv...
#   intersphinx inventory has moved: https://docs.python.org/ -> https://docs.python.org/3
#   [autosummary] generating autosummary for: administration.rst, api/remoteappmanager.application.rst, api/remoteappmanager.cli.remoteappdb.__main__.rst, api/remoteappmanager.cli.remoteapprest.__main__.rst, api/remoteappmanager.command_line_config.rst, api/remoteappmanager.db.csv_db.rst, api/remoteappmanager.db.interfaces.rst, api/remoteappmanager.db.orm.rst, api/remoteappmanager.docker.async_docker_client.rst, api/remoteappmanager.docker.container.rst, ..., design.rst, developer.rst, developer/ci_instructions.rst, developer/docker.rst, index.rst, troubleshoot/database.rst, troubleshoot/docker.rst, troubleshoot/general.rst, troubleshooting.rst, utilities.rst
#   building [mo]: targets for 0 po files that are out of date
#   building [html]: targets for 40 source files that are out of date
#   updating environment:
#   40 added, 0 changed, 0 removed
#   reading sources... [  2%] administration
#   reading sources... [  5%] api/remoteappmanager.application
# Loaded extensions:
#   sphinx.ext.autodoc (1.4.4) from /home/pzahemszky/Repositories/simphony-remote/venv/lib/python3.6/site-packages/sphinx/ext/autodoc.py
#   sphinx.ext.napoleon (1.4.4) from /home/pzahemszky/Repositories/simphony-remote/venv/lib/python3.6/site-packages/sphinx/ext/napoleon/__init__.py
#   sphinx.ext.intersphinx (1.4.4) from /home/pzahemszky/Repositories/simphony-remote/venv/lib/python3.6/site-packages/sphinx/ext/intersphinx.py
#   sphinx.ext.todo (1.4.4) from /home/pzahemszky/Repositories/simphony-remote/venv/lib/python3.6/site-packages/sphinx/ext/todo.py
#   sphinx.ext.coverage (1.4.4) from /home/pzahemszky/Repositories/simphony-remote/venv/lib/python3.6/site-packages/sphinx/ext/coverage.py
#   sphinx.ext.viewcode (1.4.4) from /home/pzahemszky/Repositories/simphony-remote/venv/lib/python3.6/site-packages/sphinx/ext/viewcode.py
#   traitlet_documenter (unknown version) from /home/pzahemszky/Repositories/simphony-remote/venv/lib/python3.6/site-packages/traitlet_documenter/__init__.py
#   _extensions (1.4.4) from /home/pzahemszky/Repositories/simphony-remote/doc/source/_extensions/__init__.py
#   alabaster (0.7.12) from /home/pzahemszky/Repositories/simphony-remote/venv/lib/python3.6/site-packages/alabaster/__init__.py
Traceback (most recent call last):
  File "/home/pzahemszky/Repositories/simphony-remote/venv/lib/python3.6/site-packages/sphinx/cmdline.py", line 244, in main
    app.build(opts.force_all, filenames)
  File "/home/pzahemszky/Repositories/simphony-remote/venv/lib/python3.6/site-packages/sphinx/application.py", line 298, in build
    self.builder.build_update()
  File "/home/pzahemszky/Repositories/simphony-remote/venv/lib/python3.6/site-packages/sphinx/builders/__init__.py", line 251, in build_update
    'out of date' % len(to_build))
  File "/home/pzahemszky/Repositories/simphony-remote/venv/lib/python3.6/site-packages/sphinx/builders/__init__.py", line 265, in build
    self.doctreedir, self.app))
  File "/home/pzahemszky/Repositories/simphony-remote/venv/lib/python3.6/site-packages/sphinx/environment.py", line 564, in update
    self._read_serial(docnames, app)
  File "/home/pzahemszky/Repositories/simphony-remote/venv/lib/python3.6/site-packages/sphinx/environment.py", line 584, in _read_serial
    self.read_doc(docname, app)
  File "/home/pzahemszky/Repositories/simphony-remote/venv/lib/python3.6/site-packages/sphinx/environment.py", line 737, in read_doc
    pub.publish()
  File "/home/pzahemszky/Repositories/simphony-remote/venv/lib/python3.6/site-packages/docutils/core.py", line 219, in publish
    self.apply_transforms()
  File "/home/pzahemszky/Repositories/simphony-remote/venv/lib/python3.6/site-packages/docutils/core.py", line 200, in apply_transforms
    self.document.transformer.apply_transforms()
  File "/home/pzahemszky/Repositories/simphony-remote/venv/lib/python3.6/site-packages/docutils/transforms/__init__.py", line 171, in apply_transforms
    transform.apply(**kwargs)
  File "/home/pzahemszky/Repositories/simphony-remote/venv/lib/python3.6/site-packages/sphinx/transforms.py", line 170, in apply
    apply_source_workaround(n)
  File "/home/pzahemszky/Repositories/simphony-remote/venv/lib/python3.6/site-packages/sphinx/util/nodes.py", line 51, in apply_source_workaround
    for classifier in reversed(node.parent.traverse(nodes.classifier)):
TypeError: 'generator' object is not reversible
```

</details>

See e.g. https://github.com/simphony/simphony-remote/runs/4254631534.

--- a/doc-requirements.txt
+++ b/doc-requirements.txt
@@ -1,3 +1,6 @@
 sphinx==1.4.4
+# docutils version 0.18 is known to cause issues with Sphinx
+# See https://github.com/sphinx-doc/sphinx/issues/9727
+docutils<0.18
 astor==0.8.1
 git+https://github.com/simphony/traitlet-documenter.git@v0.1.0#egg=traitlet_documenter

